File: app/services/auth-service/src/routes/health.ts
import type { FastifyInstance } from 'fastify';

export async function healthRoutes(app: FastifyInstance) {
  app.get('/live', async () => ({ status: 'ok', service: 'auth-service' }));
  app.get('/ready', async () => ({ status: 'ready', service: 'auth-service' }));
}

File: app/services/ai-service/app/routers/health.py
from fastapi import APIRouter

router = APIRouter(prefix="/health", tags=["health"])


@router.get("/live")
async def liveness():
    return {"status": "ok", "service": "ai-service"}


@router.get("/ready")
async def readiness():
    ready = True
    details: dict[str, str] = {}

    try:
        import pytesseract

        pytesseract.get_tesseract_version()
        details["tesseract"] = "available"
    except Exception:
        details["tesseract"] = "unavailable (mock mode)"

    return {
        "status": "ready" if ready else "not ready",
        "service": "ai-service",
        "details": details,
    }
